src/gluonts/dataset/loader.py

```diff
@@ -11,186 +11,270 @@
+import itertools
+import random
 import logging
-from typing import Iterable, Iterator, Optional
+from multiprocessing import Process, Manager, Queue
+from multiprocessing.reduction import ForkingPickler
+import io
+import pickle
+import sys
+from queue import Empty
 
-# Third-party imports
-import mxnet as mx
-import numpy as np
-
-# First-party imports
-from gluonts.core.component import DType
-from gluonts.dataset.common import DataBatch, DataEntry, Dataset
-from gluonts.dataset.parallelized_loader import ParallelDataLoader
+from gluonts.dataset.common import DataEntry, DataBatch, Dataset
+from gluonts.dataset.util import MPWorkerInfo, batcher
 from gluonts.transform import Transformation
+from gluonts.transform.dataset import TransformedDataset
+
+
+logger = logging.getLogger(__name__)
 
 
-class DataLoader(Iterable[DataEntry]):
+class CyclicIterable(Iterable):
+    def __init__(self, base_iterable: Iterable) -> None:
+        self.base_iterable = base_iterable
+
+    def __iter__(self):
+        while True:
+            yield from self.base_iterable
+
+
+class PseudoShuffledIterator(Iterator):
     """
-    An abstract Iterable type for iterating and transforming a dataset,
-    in batches of a prescribed size.
-
-    Parameters
-    ----------
-    dataset
-        The dataset from which to load data.
-    transform
-        A transformation to apply to each entry in the dataset.
-    batch_size
-        The size of the batches to emit.
-    ctx
-        MXNet context to use to store data.
-    dtype
-        Floating point type to use.
-    num_workers
-        The number of multiprocessing workers to use for data preprocessing.
-        By default 0, in which case no multiprocessing will be utilized.
-    num_prefetch
-        The number of prefetching batches only works if `num_workers` > 0.
-        If `prefetch` > 0, it allow worker process to prefetch certain batches before
-        acquiring data from iterators.
-        Note that using large prefetching batch will provide smoother bootstrapping performance,
-        but will consume more shared_memory. Using smaller number may forfeit the purpose of using
-        multiple worker processes, try reduce `num_workers` in this case.
-        By default it defaults to `num_workers * 2`.
-    cyclic
-        Indicates whether the dataset is traversed potentially multiple times.
-    shuffle_buffer_length
-        The length of the buffer used to do pseudo shuffle.
-        If not None, the loader will perform pseudo shuffle when generating batches.
-        Note that using a larger buffer will provide more randomized batches, but will make the job require a bit
-        more time to be done.
+    A wrapper class which takes a serialized iterator as an input and generates a
+    pseudo randomized iterator using the same elements from the input iterator.
     """
 
+    def __init__(self, base_iterator: Iterator, shuffle_buffer_length: int):
+        self.shuffle_buffer: list = []
```

Comment:
```suggestion
        self.shuffle_buffer = []
```

Reply:
This was to make mypy happy

Reply:
I think mypy might have complained because it didn't know what the containing type is. With `list` probably indicate `List[Any]`.